app/src/main/java/com/eveningoutpost/dexdrip/Models/NSClientChat.java

```diff
@@ -48,7 +48,8 @@ public void run() {
                         }
                     }
 
-                    data.put("insulin", thistreatment.insulin);
+                    data.put("insulin", thistreatment.insulinSummary);
+                    data.put("insulinInjections", thistreatment.insulinJSON);
```

Comment:
Does nightscout understand this data for `insulinInjections` ?

Reply:
yes - nightscout (tested since version 0.11) will just "use" the insulinSummary for its predictions BUT it will store the complete treatments injection as a JSON structure under key "insulinInjections" in its MongoDB. When reading treatments from nightscout (follower mode) the complete injection is recovered from the stored JSON string.

To let nightscout really use the complete injection (as in xdrip+) we have to implement a kind of profile sync service between NS and xdrip+ but this will rise a bigger change in NS